api/handlers.go

```diff
@@ -38,35 +37,42 @@ func (server *Server) ParseDockerCommand(w http.ResponseWriter, r *http.Request)
 		}
 		log.Msgf("Returned %d in %v", code, time.Since(start))
 	}()
-	err := json.NewDecoder(r.Body).Decode(&dockerCmd)
 
+	err := json.NewDecoder(r.Body).Decode(&dockerCmds)
 	if err != nil {
 		errorMsg = fmt.Sprintf("Error decoding request body: %v", err)
 		code = http.StatusBadRequest
 		return
 	}
 
-	// Validate the command.
-	if dockerCmd.Command == "" {
-		errorMsg = "Docker command cannot be empty"
-		code = http.StatusBadRequest
-		return
-	}
+	var p *parser.Parser
+	for _, cmd := range dockerCmds.Commands {
+		if cmd == "" {
+			errorMsg = "Docker command cannot be empty"
+			code = http.StatusBadRequest
+			return
+		}
 
-	// Create a new Parser
-	p, err := parser.New(dockerCmd.Command)
-	if err != nil {
-		errorMsg = fmt.Sprintf("Error creating parser: %v", err)
-		code = http.StatusBadRequest
-		return
-	}
+		// Create a new Parser or append to existing parser
+		if p == nil {
+			p, err = parser.New(cmd)
+		} else {
+			yamlBytes := []byte(p.String())
+			p, err = parser.AppendToYAML(yamlBytes, cmd)
+		}
 
-	// Parse the Docker command
-	err = p.Parse()
-	if err != nil {
-		errorMsg = fmt.Sprintf("Error parsing Docker command: %v", err)
```

Comment:
Several errorMsg and code setting occurred. Is it possible to encapsulate that in a single method and use the method instead. It feels like code repetition with changes in only the message.